### PR TITLE
Updated `Breadcrumb` back button styling

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-summer-editions/issues/947 by overriding the styles of `Button` when in the  Page Header. This follows the same solution that the other buttons in the page [secondary actions button groups use](https://github.com/Shopify/polaris/blob/4d018664742bc3c9e7c4b577d31dcd05d09a2a6f/polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss#L8)

### WHAT is this pull request doing?

https://github.com/Shopify/polaris/assets/4642404/2ecc28a3-f320-4ea7-aa41-e84f07db33c8

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Breadcrumbs/Breadcrumbs.tsx
Comment (yurm04) [inline polaris-react/src/components/Breadcrumbs/Breadcrumbs.tsx:23]: probably missing context here but can we just get rid of this and just use the `Button` component?

File: polaris-react/src/components/Page/components/Header/Header.scss
Comment (yurm04) [inline polaris-react/src/components/Page/components/Header/Header.scss:21]: The [other buttons in the page header are styled with overrides](https://github.com/Shopify/polaris/blob/4d018664742bc3c9e7c4b577d31dcd05d09a2a6f/polaris-react/src/components/ActionMenu/components/SecondaryAction/SecondaryAction.scss#L8) so I matched that pattern but open to another suggestion as well

File: polaris-react/src/components/Page/components/Header/Header.scss
Comment (yurm04) [inline polaris-react/src/components/Page/components/Header/Header.scss:37]: I think I need `!important` here because the focus state for `Button` is highly specific. This is also done in the other buttons

File: polaris-react/src/types.ts
Comment (yurm04) [inline polaris-react/src/types.ts]: is this cool for typescript? added it because above using the `Button` component and passing a function with an argument to `onPointerDown`

Comment (sam-b-rose) [reply polaris-react/src/types.ts]: It is fine to update the typing here, but it would be better if we could improve the naming and not use `any`.

Maybe try using something like `MouseEvent<'button'>`? I'm not sure of the exact types but can help jam on this if TS is giving you a lot of push back

File: polaris-react/src/components/Page/components/Header/Header.scss
Comment (sam-b-rose) [reply polaris-react/src/components/Page/components/Header/Header.scss:37]: Not ideal, but I think this solution could be refactored more systematically across the other areas as well later on.

For now, this is a "patterned fix" for this type of issue 👍

**Review by sam-b-rose — APPROVED** [review]
The fix looks good! Though, the solution doesn't seem very scalable even though we've used it in another area for fixing a similar issue 🤔 

Instead of styling the `.BreadcrumbWrapper` in the `Page` component SCSS, maybe we could improve the `se23` overrides in the breadcrumb component itself.

No blockers here, just pondering other alternatives. Happy to get this fix shipped now 🚢

File: polaris-react/src/components/Breadcrumbs/Breadcrumbs.tsx
Comment (sam-b-rose) [reply polaris-react/src/components/Breadcrumbs/Breadcrumbs.tsx:23]: I would think so as well. The `Breadcrumbs` use to support multiple action items, but now it only accepts one. We should rethink how this component fits into the system overall

**Comment by yurm04** [comment]
thanks @samrose3 for the feedback, i'm going to merge this in after a UX 👍 then create an issue to track your comments. I thought I could quickly tackle the TS feedback above but couldn't figure it out so let's definitely pair so I can learn how it's done 🙌

**Review by heyjoethomas — APPROVED** [review]

File: polaris-react/src/types.ts
Comment (sam-b-rose) [reply polaris-react/src/types.ts]: Let's hold off on merging this until we have discussed the larger implications of changing this typing. @aaronccasanova made a great point that changing this to `any` will impact our users and could make it difficult for us in the future reversing or changing this decision

**Review by sam-b-rose — CHANGES_REQUESTED** [review]

File: polaris-react/src/types.ts
Comment (aaronccasanova) [reply polaris-react/src/types.ts]: Pushed up a fix with the correct event type: https://github.com/Shopify/polaris/pull/9733/commits/2d8ebbbd99a348214f5fc3fd244cd710642ee864

I think we should snapshot this PR and make sure Web CI passes as this change has the potential to break builds.

**Review by sam-b-rose — APPROVED** [review]

**Review by sarahill — CHANGES_REQUESTED** [review]
@yurm04 the back button should still be transparent by default. Looks like it's getting a background color now.

expected
image

actual
image

**Comment by yurm04** [comment]
Screenshot 2023-07-19 at 12 59 09 PM

@sarahill thanks for catching this! button is now transparent by default

**Review by sarahill — APPROVED** [review]

File: .changeset/fifty-bats-visit.md
Comment (kyledurand) [inline .changeset/fifty-bats-visit.md:5]: @yurm04 can you revert this? Create a new PR and delete this empty file?

**Pull request merged**
